floodviz/tests/test_utils.py

```diff
@@ -0,0 +1,210 @@
+                           '# Provisional data are subject to revision. Go to\n' \
+                           '# http://help.waterdata.usgs.gov/policies/provisional-data-statement for more information.\n' \
+                           '#\n' \
+                           '# File-format description:  http://help.waterdata.usgs.gov/faq/about-tab-delimited-output\n' \
+                           '# Automated-retrieval info: http://help.waterdata.usgs.gov/faq/automated-retrievals\n' \
+                           '#' \
+                           '# Contact:   gs-w_support_nwisweb@usgs.gov\n' \
+                           '# retrieved: 2017-07-03 14:46:35 -04:00 (natwebvaas01)\n' \
+                           '#\n' \
+                           '# Data for the following 1 site(s) are contained in this file\n' \
+                           '#    USGS 05481950 Beaver Creek near Grimes, IA\n' \
+                           '# -----------------------------------------------------------------------------------\n' \
+                           '#\n' \
+                           '# TS_ID - An internal number representing a time series.\n' \
+                           '# IV_TS_ID - An internal number representing the Instantaneous Value time series from which the daily' \
+                           'statistic is calculated.\n' \
+                           '#\n' \
+                           '# Data provided for site 05481950\n' \
+                           '#    TS_ID       Parameter    Statistic  IV_TS_ID       Description\n' \
+                           '#    43051       00060        00003      44227          Discharge, cubic feet per second (Mean)\n' \
+                           '#\n' \
+                           '# Data-value qualification codes included in this output:\n' \
+                           '#     A  Approved for publication -- Processing and review completed.\n' \
+                           '#\n' \
+                           'agency_cd\tsite_no\tdatetime\t43051_00060_00003\t43051_00060_00003_cd\n' \
+                           '5s      15s     20d     14n     10s\n' \
+                           'USGS\t05481950\t2008-07-05\t430\tA'
+
+    def test_bad_request(self):
+        with requests_mock.Mocker() as m:
+            m.get(self.mock_url_peak, status_code=404)
+            self.assertEqual(parse_rdb(self.endpoint_mock, self.test_dict_peak), None)
+
+    def test_valid_data_peak(self):
+        with requests_mock.Mocker() as m:
+            m.get(self.mock_url_peak, text=self.response_peak)
+            self.assertEqual(parse_rdb(self.endpoint_mock, self.test_dict_peak), self.valid_return_peak)
+
+    def test_valid_data_dv(self):
+        with requests_mock.Mocker() as m:
+            m.get(self.mock_url_dv, text=self.response_dv)
+            self.assertEqual(parse_rdb(self.endpoint_mock, self.test_dict_dv), self.valid_return_dv)
+
+
+class TestRdbParseMap(unittest.TestCase):
```

Comment:
The unit test should only test the particular unit, not how it will be used in the application. The unit test does not need to reflect a "real" example but should mimic the important pieces. In this case all we need to do is test whether it parses a file in rdb format, not necessarily an actual response.